spec/message.abnf

```diff
@@ -20,12 +20,14 @@ variant = key *(s key) [s] quoted-pattern
 key = literal / "*"
 
 expression = literal-expression / variable-expression / function-expression
+           / private-use-expression / reserved-expression
 literal-expression = "{" [s] literal [s annotation] [s] "}"
 variable-expression = "{" [s] variable [s annotation] [s] "}"
-function-expression = "{" [s] annotation [s] "}"
-annotation = (function *(s option))
-           / reserved-annotation
-           / private-use-annotation
+function-expression = "{" [s] function-annotation [s] "}"
+private-use-expression = "{" [s] private-use-annotation [s] "}"
+reserved-expression = "{" [s] reserved-annotation [s] "}"
+annotation = function-annotation / private-use-annotation / reserved-annotation
+function-annotation = function *(s option)
```

Comment:
Where did this come from?

I like the existing tight definition of `annotation` and am not very thrilled by lots of different expression types. The syntax doesn't need the proliferation of expressions. The data model (above) captures the deeper split between functions, private-use (maybe supported), and reserved (unsupported) and that is an outcome of parsing the syntax. But we don't, I guess, need to go all-in with every kind of placeholder. THoughts?

Reply:
We don't _need_ to make this change, but if we don't then the less direct translation from ABNF to data model can be unnecessarily confusing. Is that worth having fewer productions in the grammar?
```abnf
expression = literal-expression / variable-expression / nullary-expression
literal-expression = "{" [s] literal [s annotation] [s] "}"
variable-expression = "{" [s] variable [s annotation] [s] "}"
nullary-expression = "{" [s] annotation [s] "}"
annotation = (function *(s option))
           / reserved-annotation
           / private-use-annotation
```

Reply:
The (normative!) grammar is **_way_** more important to me than the (informative) data-model. It's important that the grammar be easily understood and as conceptually clear as possible (using Einstein's simplicity maxim--_as simple as possible but no simpler_). 

Reserved annotations don't really matter--no one can use them. The private use ones are where any additional heat comes into play. The nature of how we set up private-use is that, while they might be "function-like" they don't have to be.

I don't think `nullary-expression` is as clean as `function-expression` or maybe `annotation-expression`?